src/main/java/com/mathworks/ci/AddMatlabToPathBuildWrapper.java

```diff
@@ -0,0 +1,155 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2019-2020 The MathWorks, Inc.
+ * 
+ * This class is BuildWrapper which accepts the "matlabroot" from user and updates the PATH varible with it.
+ * which could be later used across build.
+ * 
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildWrapperDescriptor;
+import hudson.util.FormValidation;
+import hudson.util.FormValidation.Kind;
+import jenkins.tasks.SimpleBuildWrapper;
+
+public class AddMatlabToPathBuildWrapper extends SimpleBuildWrapper {
+
+    private String matlabRootFolder;
+    private EnvVars env;
+
+    @DataBoundConstructor
+    public AddMatlabToPathBuildWrapper() {}
+
+    public String getMatlabRootFolder() {
+        return this.matlabRootFolder;
+    }
+
+    @DataBoundSetter
+    public void setMatlabRootFolder(String matlabRootFolder) {
+        this.matlabRootFolder = matlabRootFolder;
+    }
+
+    private String getLocalMatlab() {
+        return this.env == null ? getMatlabRootFolder() : this.env.expand(getMatlabRootFolder());
+    }
+
+    private void setEnv(EnvVars env) {
+        this.env = env;
+    }
+
+    @Symbol("Matlab")
+    @Extension
+    public static final class MatlabBuildWrapperDescriptor extends BuildWrapperDescriptor {

```

Comment:
Descriptor class name needs to be updated to match the new Wrapper name.

Reply:
Done.